liblouis/compileTranslationTable.c

```diff
@@ -4582,13 +4587,21 @@ resolveSubtable(const char *table, const char *base, const char *searchPath) {
 			last = (*cp == '\0');
 			*cp = '\0';
 			if (dir == cp) dir = ".";
+			if (strlen(dir) + strlen(table) + 1 >= MAXSTRING * sizeof(char) * 2) {
+				free(searchPath_copy);
+				goto failure;
+			}
 			sprintf(tableFile, "%s%c%s", dir, DIR_SEP, table);
 			if (stat(tableFile, &info) == 0 && !(info.st_mode & S_IFDIR)) {
 				_lou_logMessage(LOU_LOG_DEBUG, "found table %s", tableFile);
 				free(searchPath_copy);
 				return tableFile;
 			}
 			if (last) break;
+			if (strlen(dir) + strlen(table) + 16 >= MAXSTRING * sizeof(char) * 2) {
```

Comment:
The magic number is a bit hard to read (say in a year when I'm trying to understand what is going on). Can you replace it with something like `strlen("liblouis")` or at least add a comment what 16 stands for?

Reply:
Now I parse it into `strlen(dir) + strlen("liblouis") + strlen("tables") + strlen(table) + 3`.

Reply:
Excellent